### PR TITLE
MPDX-7362: EMF Report Date Selector

### PR DESCRIPTION
**Pull request by dbenton9**
To add a date selector I had to get a little fancy. The `tag_histories` report doesn't accept a traditional date range (to my knowledge). Instead of a normal date range, it accepts an integer and a letter: d (day), m (month), y(year). For example `4m`, this tells the report to go pull the past 4 months of data. The report also pulls start and end dates for each month and these dates are used by this code base to make a followup call for tasks completed within those dates.

So I was left manipulating the integer letter combo. I did this by writing a method (`number_of_intervals_between_now_and_target_date`) to calculate the number of months between the current month and desired month so I knew what the integer needed to be in the URL. This then provided me with the data to display in the score cards.

### CODE REVIEW

File: app/lib/account_list_stats_loader.rb [hunk 1]
```diff
@@ -45,8 +45,8 @@ def zip_tags_report(data, tags_report)
     data
   end
 
-  def tags_report(type)
-    range = "#{number_of_time_periods(type)}#{type == :weekly ? "w" : "m"}"
+  def tags_report(type, target_date)
+    range = "#{number_of_intervals_between_now_and_target_date(type, target_date)}#{type == :weekly ? "w" : "m"}"
```
Comment (dbenton9) [inline app/lib/account_list_stats_loader.rb:49]: For this particular report I need to tell it how many months back I want data from so this method `number_of_intervals_between_now_and_target_date` calculates that and then passes in an integer value.

File: app/lib/account_list_stats_loader.rb [hunk 2]
```diff
@@ -56,6 +56,20 @@ def tags_report(type)
...
 
+  def number_of_intervals_between_now_and_target_date(type, target_date)
+    # This calculates # of months between today and target_date (1 added at end because report includes current month)
+    target_difference = (Date.today.year * 12 + Date.today.month) - (target_date.year * 12 + target_date.month) + 1
+
+    case type
+    when :weekly
+      5
+    when :monthly
+      (target_difference if target_difference > 5) || 5
```
Comment (dbenton9) [inline app/lib/account_list_stats_loader.rb:67]: For monthly reports we're going to report on at least 5 months worth of data so we don't want less than 5.

**Review by jbirdjavi — APPROVED** [review]
I think this looks good. I should probably pull it in and play with it locally, but I'm also very lazy today...

**Comment by dbenton9** [comment]
@jbirdjavi, we'll just let the users play with it, break it, and then I can fix it! 😄

**Comment by jbirdjavi** [comment]
> @jbirdjavi, we'll just let the users play with it, break it, and then I can fix it! 😄

Seems like a good plan for this app at least. :)

**Pull request merged**